src/imap/client.rs

```diff
@@ -95,9 +96,15 @@ impl Client {
         domain: &str,
         strict_tls: bool,
     ) -> Result<Self> {
-        let stream = TcpStream::connect(addr).await?;
+        let tcp_stream = TcpStream::connect(addr).await?;
+        let mut timeout_stream = TimeoutStream::new(tcp_stream);
+        timeout_stream.set_write_timeout(Some(Duration::new(IMAP_TIMEOUT, 0)));
```

Comment:
I'd generally prefer `from_secs` over `new` because `from_secs` makes it clearer that the argument is in seconds.

But since IMAP_TIMEOUT could just be directly a type-safe Duration, I created https://github.com/deltachat/deltachat-core-rust/pull/3823 instead.